testdata/p4tc_samples_outputs/global_action_example_02.c

```diff
@@ -208,10 +208,10 @@ static __always_inline int process(struct __sk_buff *skb, struct my_ingress_head
                 goto reject;
             }
 
-            hdr->ethernet.dstAddr = (u64)((load_dword(pkt, BYTES(ebpf_packetOffsetInBits)) >> 16) & EBPF_MASK(u64, 48));
+            __builtin_memcpy(&hdr->ethernet.dstAddr, pkt + BYTES(ebpf_packetOffsetInBits), 6);
             ebpf_packetOffsetInBits += 48;
 
-            hdr->ethernet.srcAddr = (u64)((load_dword(pkt, BYTES(ebpf_packetOffsetInBits)) >> 16) & EBPF_MASK(u64, 48));
+            __builtin_memcpy(&hdr->ethernet.srcAddr, pkt + BYTES(ebpf_packetOffsetInBits), 6);
```

Comment:
Is this is a portable function? I believe clang requires language extensions to support this particular memcpy version.

Reply:
Would be nice to use a less compiler-specific version here.

Reply:
__builtin_memcpy is commonly used in eBPF code (not only P4-ebpf). There are many instances in same code where __builtin_memcpy is used.